provision/infinispan/README.adoc

```diff
@@ -2,55 +2,26 @@
 
 Deploy an Infinispan cluster (with or without cross-site) and configures the necessary caches for Keycloak.
 
-== Required variables
+== Taskfile
 
-* `CLUSTER_1`: The ROSA cluster name of site 1.
-* `CLUSTER_2`: The ROSA cluster name of site 2.
-* `NS_1`: The namespace to install Infinispan in site 1.
-* `NS_2`: The namespace to install Infinispan in site 2.
+For single cluster mode:
 
-== Optional variables:
-* `KUBECONFIG_1`: Path to `KUBECONFIG` to use for CLUSTER_1.
-* `KUBECONFIG_2`: Path to `KUBECONFIG` to use for CLUSTER_2.
-
-If file does not exist, rosa_oc_login will be invoked and the configuration stored in this path.
-
-== Infinispan cluster customization:
-
-* `XSITE_MODE`: The cross-site strategy, default to SYNC.
-* `ISPN_REPLICAS`: The number of Infinispan pods.
-
-== Operation Modes
-
-* CLUSTER_1 == CLUSTER_2 and NS_1 == NS_2 -> Single Infinispan cluster without cross-site.
-
-Example:
-[source, bash]
-----
-CLUSTER_1="gh-pruivo" CLUSTER_2="gh-pruivo" NS_1="ispn-server" NS_2="ispn-server" ./create_ispn_clusters.sh
-----
-
-* CLUSTER_1 == CLUSTER_2 and NS_1 != NS_2 -> Infinispan clusters with cross-site enabled in a single OCP cluster.
-Each namespace gets an Infinispan cluster, and they are linked together with cross-site.
-
-Example:
 [source, bash]
 ----
-CLUSTER_1="gh-pruivo" CLUSTER_2="gh-pruivo" NS_1="server-site-1" NS_2="server-site-2" ./create_ispn_clusters.sh
+task single-cluster ROSA_CLUSTER_NAME=gh-keycloak OC_NAMESPACE=infinispan
 ----
 
-* CLUSTER_1 != CLUSTER_2 -> Infinispan clusters with cross-site enabled in 2 different OCP clusters.
-The namespaces must be set and they can be the same or different namespaces.
+For cross-site between different namespaces:
 
-Example:
 [source, bash]
 ----
-CLUSTER_1="gh-pruivo" CLUSTER_2="gh-keycloak" NS_1="ispn-server" NS_2="ispn-server" ./create_ispn_clusters.sh
+task crossdc-single ROSA_CLUSTER_NAME=gh-keycloak OC_NAMESPACE_1=ispn-1 OC_NAMESPACE_2=ispn-2
 ----
 
+And finally, for cross-site between different OCP clusters:
 [source, bash]
 ----
-CLUSTER_1="gh-pruivo" CLUSTER_2="gh-keycloak" NS_1="server-site-1" NS_2="server-site-2" ./create_ispn_clusters.sh
+task crossdc ROSA_CLUSTER_NAME_1=gh-keycloak ROSA_CLUSTER_NAME_2=gh-pruivo OC_NAMESPACE_1=infinispan OC_NAMESPACE_2=infinispan
```

Comment:
can we add some example log message that would reflect a successful creation of this cross-site in all three situations of.,

- single-cluster
- crossdc-single
- crossdc

Reply:
added waiting conditions. It should print the following
```
task: [wait-cluster] echo "Waiting for the Infinispan cluster to form on namespace 'pruivo1' (timeout after 300 seconds)"
[wait-cluster] Waiting for the Infinispan cluster to form on namespace 'pruivo1' (timeout after 300 seconds)
task: [wait-cluster] KUBECONFIG=".task/kubecfg/crossdcsingle" oc wait --for condition=WellFormed --timeout=300s infinispans.infinispan.org -n "pruivo1" infinispan
[wait-cluster] infinispan.infinispan.org/infinispan condition met
task: [wait-cluster] echo "Waiting for the Infinispan cluster to form on namespace 'pruivo2' (timeout after 300 seconds)"
[wait-cluster] Waiting for the Infinispan cluster to form on namespace 'pruivo2' (timeout after 300 seconds)
task: [wait-cluster] KUBECONFIG=".task/kubecfg/crossdcsingle" oc wait --for condition=WellFormed --timeout=300s infinispans.infinispan.org -n "pruivo2" infinispan
[wait-cluster] infinispan.infinispan.org/infinispan condition met
task: [wait-crossdc] echo "Waiting for the Infinispan cross-site to form on namespace 'pruivo1' (timeout after 300 seconds)"
[wait-crossdc] Waiting for the Infinispan cross-site to form on namespace 'pruivo1' (timeout after 300 seconds)
task: [wait-crossdc] KUBECONFIG=".task/kubecfg/crossdcsingle" oc wait --for condition=CrossSiteViewFormed --timeout=300s infinispans.infinispan.org -n "pruivo1" infinispan
[wait-crossdc] infinispan.infinispan.org/infinispan condition met
task: [wait-crossdc] echo "Waiting for the Infinispan cross-site to form on namespace 'pruivo2' (timeout after 300 seconds)"
[wait-crossdc] Waiting for the Infinispan cross-site to form on namespace 'pruivo2' (timeout after 300 seconds)
task: [wait-crossdc] KUBECONFIG=".task/kubecfg/crossdcsingle" oc wait --for condition=CrossSiteViewFormed --timeout=300s infinispans.infinispan.org -n "pruivo2" infinispan
[wait-crossdc] infinispan.infinispan.org/infinispan condition met
```

Reply:
Thanks for adding the wait and listing out the actual output from the task output and at the sametime, any hint from the Keycloak server on how the logs look for a successful cross-site creation @pruivo